### PR TITLE
10M entries generated in 1 CSV file

### PR DESCRIPTION
**Pull request by codysmith287**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,17 +10,22 @@
   "scripts": {
     "react-dev": "webpack -d --watch",
     "seed": "mongoimport --db bandland --collection users --file userData.json --jsonArray",
+    "gen-data": "node --max-old-space-size=8192 database/mockDataGen.js",
```
Comment (hannahsohkim) [inline package.json:13]: I would recommend committing more often. Adding a script could be a separate commit from the data generation you implemented.

File: database/mockDataGen.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+// let randomComments = generateRandomComments();
+// console.log(randomComments);
+// console.log(randomComments.slice(0, randomComments.length - 2).split(";,"));
+
+const createCSVdata = () => {
+  console.time('timing seed');
+  writer.pipe(fs.createWriteStream('comments.csv'));
+  for (let i = 0; i < 10000000; i++) {
+    const album_id = i;
```
Comment (hannahsohkim) [inline database/mockDataGen.js:31]: Is this a foreign key you are defining here?

File: database/mockDataGen.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+
+// let randomComments = generateRandomComments();
+// console.log(randomComments);
+// console.log(randomComments.slice(0, randomComments.length - 2).split(";,"));
+
+const createCSVdata = () => {
+  console.time('timing seed');
+  writer.pipe(fs.createWriteStream('comments.csv'));
+  for (let i = 0; i < 10000000; i++) {
```
Comment (hannahsohkim) [inline database/mockDataGen.js:30]: What happens when you define randomComments inside of the for loop? (i.e. move line 23 inside the for loop) and then add in the array like so ===> {comments: {randomComments}} ? does it crash/not generate?

File: package.json (hunk 1 above)
Comment (codysmith287) [reply package.json:13]: Good point! I will make sure to do so in the future. Thank you :)

File: database/mockDataGen.js (hunk 2 above)
Comment (codysmith287) [reply database/mockDataGen.js:31]: If you are referring to 'const album_id' it's just a naming convention I've been using.

File: database/mockDataGen.js (hunk 3 above)
Comment (codysmith287) [reply database/mockDataGen.js:30]: It doesn't crash or not generate, rather lines 23-25 were written for mainly for testing how I would split all the nested data (i.e. multiple comments in one string) client side in future read operations (CRUD). The main factor in whether it crashes or not generates is the number of comments that I'm randomly generating. One to five randomly generate comments for each entry of 10 million entry take 30+ minutes, only about 23 seconds for 1 M though. I don't know the exact time for 10 million with one to five random comments since I stopped it after about 30 minutes. My goal for data generation was under 10 minutes.

File: database/mockDataGen.js (hunk 2 above)
Comment (hannahsohkim) [reply database/mockDataGen.js:31]: sorry let me rephrase! is this meant to be to match up comments to album_ids (which would work as a pseudo foreign key?) just wondering since upon insert the id would be auto generated anyway if its in your schema right? esp since you are just incrementing and not assigning random numbers as album_id

File: client/dist/bundle.js [hunk 4]
```diff
@@ -94,7 +94,7 @@
...
...b3VyY2VzIjpbIndlYnBhY2s6Ly8vLi9jbGllbnQvc3JjL2NvbXBvbmVudHMvQ29tbWVudC5qc3g/YjJlNSJdLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuXG5jbGFzcyBDb21tZW50IGV4dGVuZHMgUmVhY3QuQ29tcG9uZW50IHtcbiAgY29uc3RydWN0b3IocHJvcHMpIHtcbiAgICBzdXBlcihwcm9wcylcbiAgfVxuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gKFxuICAgICAgPGRpdiBjbGFzc05hbWU9XCJjb21tZW50QmxvY2tcIj5cbiAgICAgICAgPGRpdiBjbGFzc05hbWU9XCJwaWNcIj5cbiAgICAgICAgICA8aW1nIHNyYz17dGhpcy5wcm9wcy51c2VyLnByb2ZpbGVJbWFnZVVSTH0gaGVpZ2h0PVwiMjhcIiB3aWR0aD1cIjI4XCI+PC9pbWc+XG4gICAgICAgIDwvZGl2PlxuICAgICAgICA8ZGl2IGNsYXNzTmFtZT1cIndvcmRzIGNvbW1lbnRcIj5cbiAgICAgICAgICA8YSBjbGFzc05hbWU9XCJ1c2VybmFtZVwiIGhyZWY9XCJcIj57dGhpcy5wcm9wcy51c2VyLnVzZXJuYW1lfTwvYT5cbiAgICAgICAgICB7dGhpcy5wcm9wcy51c2VyLmNvbW1lbnR9XG4gICAgICAgIDwvZGl2PlxuICAgICAgPC9kaXY+XG4gICAgKTtcbiAgfVxufVxuXG5leHBvcnQgZGVmYXVsdCBDb21tZW50O1xuIl0sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7QUFBQTtBQUNBO0FBQ0E7Ozs7O0FBQ0E7QUFBQTtBQUNBO0FBREE7QUFFQTtBQUNBOzs7QUFDQTtBQUNBO0FBQ0E7QUFBQTtBQUNBO0FBQUE7QUFDQTtBQUFBO0FBQUE7QUFBQTtBQUVBO0FBQUE7QUFDQTtBQUFBO0FBQUE7QUFLQTs7OztBQWpCQTtBQUNBO0FBbUJBIiwic291cmNlUm9vdCI6IiJ9\n//# sourceURL=webpack-internal:///./client/src/components/Comment.jsx\n");
```
Comment (Druthi) [inline client/dist/bundle.js:97]: Wouldn't it be better to gitignore this file?

File: database/mockDataGen.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    const album_id = i;
+    const commentsObj = JSON.stringify({
+      username: faker.internet.userName(),
+      comment: faker.lorem.sentence(),
+      profileImgUrl: faker.internet.avatar()
+    });
+    writer.write({
+      album_id,
+      commentsObj
```
Comment (Druthi) [inline database/mockDataGen.js:39]: Hey, so are you adding the entire object as a string into the table and then handling it upon querying it in your api calls? Would that not defeat the purpose of using tables?

File: client/dist/bundle.js (hunk 4 above)
Comment (codysmith287) [reply client/dist/bundle.js:97]: Yes, it would be. Not sure why this got pushed up since I have in my .gitignore file. I'll double check to make sure. Good catch!

File: database/mockDataGen.js (hunk 5 above)
Comment (codysmith287) [reply database/mockDataGen.js:39]: Yeah, that is my intent for now. If I'm using albums as my primary items for handling 10M different endpoints that mean with 1 to 5 comments average to about 30M entries in my database which I'd need to optimize further for.

File: database/mockDataGen.js (hunk 2 above)
Comment (codysmith287) [reply database/mockDataGen.js:31]: Oh ok, I see what you're saying! I'll look into this further. Thanks.

File: database/mockDataGen.js (hunk 5 above)
Comment (therobinkim) [reply database/mockDataGen.js:39]: why not `write.write({ album_id, username, comment, profileImageUrl })` instead?

File: package.json [hunk 6]
```diff
@@ -10,17 +10,22 @@
...
   },
   "dependencies": {
     "body-parser": "^1.18.3",
     "cors": "^2.8.5",
+    "csv-write-stream": "^2.0.0",
     "express": "^4.16.4",
+    "faker": "^4.1.0",
     "grunt": "^1.0.4",
     "jquery": "^3.3.1",
+    "knex": "^0.16.5",
```
Comment (therobinkim) [inline package.json:26]: were `knex` and` pg` being used somewhere in this codebase? I notice that they're being added as packages, but I don't think I saw them being used in this commit

Comment (codysmith287) [reply package.json:26]: I am using pg now but not knex. I went ahead and uninstalled knex. Thanks for catching that.
